core/src/marker/markerManager.cpp

```diff
@@ -297,6 +298,27 @@ bool MarkerManager::buildStyling(Marker& marker) {
     try {
         // Update the draw rule for the marker.
         YAML::Node node = YAML::Load(marker.stylingString());
+
+        // Check if marker styling defines a layer in the yaml scene
+        if (Node layer = node["layer"]) {
+            if (!layer.IsScalar()) {
+                LOGE("layer for marker styling needs to be a scalar value.");
+                return false;
+            }
+
+            const DataLayer* sceneLayer = m_scene->findLayer(layer.Scalar());
```

Comment:
Restricting ourselves to top-level scene layers here is pretty limiting. We can borrow from our 'scene update' syntax for a way to specify fully-qualified layer names, like `roads.highway.link`. 

When styling the marker with a layer, we would then need to split the fully-qualified name on `.` and follow the layer tree recursively (either here or in `Scene::findLayer`).